### PR TITLE
Added data integrity tests to HDFS

### PR DESCRIPTION
**Pull request by NimaVaziri**
Mostly done, still a bit WIP.

### CODE REVIEW

**Comment by susanxhuynh** [comment]
- Removed the hardcoded JRE and Hadoop versions from shell commands.
- Replaced sleep() with polling.
- Fixed a bug in the janitor cleanup command.

File: frameworks/hdfs/integration/tests/test_data_integrity.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    delete_command = "./bin/hdfs dfs -rm -r /{}".format(filename)
+    rc, _ = run_hdfs_command(host, delete_command)
+    return rc
+
+
+def run_hdfs_command(host, command):
+    """
+    Go into the Data Node hdfs directory, set JAVA_HOME, and execute the command.
+    """
+    java_home = find_java_home(host)
```
Comment (nickbp) [inline frameworks/hdfs/integration/tests/test_data_integrity.py]: Should be able to just have something like this, without needing to run commands on the agent at all. The `framework_id` check ensures that we select the right data node in the event multiple `hdfs` instances are installed (or in the event some other service has an executor named 'data'):

```
framework_id = shakedown.get_service_framework_id("hdfs")
[...]
"JAVA_HOME=$(ls -d /var/lib/mesos/slave/slaves/*/frameworks/{}/executors/data__*/runs/latest/jre*/)".format(framework_id)
```

The `ls -d` command will be run on the agent system and will resolve the correct path.

File: frameworks/hdfs/integration/tests/test_data_integrity.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+    # Find hdfs home directory by looking up the Data Node process.
+    # Hdfs directory is found in an arg to the java command.
+    hdfs_dir_cmd = """ps -ef | grep hdfs | grep DataNode \
```
Comment (nickbp) [inline frameworks/hdfs/integration/tests/test_data_integrity.py]: Similar to the above, this lookup could be something like:
```
cd $(ls -d /var/lib/mesos/slave/slaves/*/frameworks/{}/executors/data__*/runs/latest/hadoop-*/)".format(framework_id)
```
(this gives us the explicit framework id filtering as above)

File: frameworks/hdfs/universe/config.json [hunk 3]
```diff
@@ -10,8 +10,13 @@
           "type":"string",
           "default":"hdfs"
         },
+        "role": {
```
Comment (nickbp) [inline frameworks/hdfs/universe/config.json]: Does this need a corresponding change in marathon.json.mustache?

File: frameworks/hdfs/integration/tests/test_data_integrity.py (hunk 1 above)
Comment (susanxhuynh) [inline frameworks/hdfs/integration/tests/test_data_integrity.py]: I tried to implement this suggestion but the "ls -d" returns multiple directories after the DataNode restart. I will leave the command as is.

**Pull request merged**